### ISSUE TITLE
Fire drills don't show charges

### ISSUE THREAD

**Issue by OnTheDouble** [CleverRaven/Cataclysm-DDA#56219]
### Describe the bug

A bow fire drill acts the same as a disposable lighter in-game. Instead of butane it has 20 unnamed "charges". But what's different is you can't see the charges even when the drill becomes unusable. 

There's no way to see charges, no way to recharge a spent fire drill, no way to tell between a new drill and an old one.

Same for pump fire drill.

### Steps to reproduce

1. Create 2 bow fire drills.
2. Use one of them 20 times.
3. Both fire drills look identical yet one is spent and the other is new.
4. (Bonus step). Try to [r]eload. The spent fire drill will show "no charges available"
![firedrill2](https://user-images.githubusercontent.com/82908191/159129643-e46dec87-2735-42c6-94dc-f52113a5a018.png)
![firedrill1](https://user-images.githubusercontent.com/82908191/159129645-1902570e-44bb-45db-9e89-363e7de7c317.png)

### Expected behavior

The lighter has butane charges. Those are single-use. All fire drills should show charges as well.

### Versions and configuration

2022-03-19-0915 Experimental

**Comment by Drew4484** [comment]
I believe this is intended behavior.

**Comment by OnTheDouble** [comment]
This leads to player confusion. All other items with charges have charges clearly visible (markers, batteries, filters, magazines, etc.). Thus the reasonable expectation is to show bow drill charges, too. 

How is a bow fire drill different.

**Comment by Croisis** [comment]
That is how all of the items with internal charges behave I believe so I wouldn't call it a bug per se but I do agree it is not very intuitive or user friendly in this case.

The lighter has a reloadable pocket (magazine) with up to 100 charges of butane, so you see it as "lighter (100/100 butane)".
Bow fire drill and its larger version the pump fire drill don't have any magazine containing usable material, you are using the item itself, its "durability" so to speak, like a burning torch or candle for example.

Which brings me to my idea... What if we added the BURNOUT flag to both drills?
You still won't be able to tell exactly how many charges it has left (which is intended) but you will have a rough idea at least.
The wording displayed is not !00% accurate but good enough I think as you technically ARE burning the bow itself...

Fuel: It's new, and ready to burn.
Fuel: Almost new, with much material to burn.
Fuel: More than a quarter has burned away.
Fuel: More than half has burned away.
Fuel: Less than a quarter left to burn.
Fuel: Almost completely burned out.

**Comment by OnTheDouble** [comment]
A rough estimate would be cool. Or maybe just a durability thing like in a weapon? The drill can even break after the last use. Whichever works. My character can see how burnt the plank he is using is after all.

This will eliminate the question why something without [visible] charges suddenly stops working.

**Comment by Croisis** [comment]
This is how it looks ingame, it is an existing flag used for actually burning items, like the mentioned torch or candle.
![image](https://user-images.githubusercontent.com/101677039/159136479-091df14f-f84d-486e-83dc-710139e43244.png)

**Comment by OnTheDouble** [comment]
I feel like showing charges directly is easier to see because you don't have to go into [e]xamine mode — like for a candle [38/100].

**Comment by Croisis** [comment]
Heh, well it sure would be more readable but the candle actually has a pocket with candle wax, nor reloadable or unloadable, but still a pocket with actual material. 
Which is kinda interesting when I think about it since when you (e)xamine it, you can see both the actual charges from the handling of the pocket with candle_wax and the rough estimate of the fuel left from the BURNOUT flag... Not confusing at all.

**Comment by OnTheDouble** [comment]
Indeed. Thanks for looking into my issue! I appreciate it.

**Comment by Croisis** [comment]
Well since I'm just starting to contribute here I'm a bit reluctant to just make the call in this case myself TBH.
I think the flag would make more sense in this case than actual pocket with visible charges. 
Simply because it is stoneage tech and just a rough estimate reflects that nicely IMHO.
On the other hand the flag is clearly intended for items that are actually on fire so it would be a little whacky fix I guess.

@Drew4484 What do you think? Is it a good idea or a no-no to fix it like that?

**Comment by OnTheDouble** [comment]
I hope QOL concerns win because it's better from UX viewpoint.

**Comment by Drew4484** [comment]
> Well since I'm just starting to contribute here I'm a bit reluctant to just make the call in this case myself TBH.
> I think the flag would make more sense in this case than actual pocket with visible charges. 
> Simply because it is stoneage tech and just a rough estimate reflects that nicely IMHO.
> On the other hand the flag is clearly intended for items that are actually on fire so it would be a little whacky fix I guess.
> 
> @Drew4484 What do you think? Is it a good idea or a no-no to fix it like that?

If a real life version can be visually inspected to show remaining uses, then that's fine for our purposes.

**Comment by OnTheDouble** [comment]
![bowDrill](https://user-images.githubusercontent.com/82908191/159138226-4b8a1553-7045-434f-8499-02d125b33ee6.jpg)
I believe you can roughtimate how many notches you can make until there's no more space on that plank.

**Comment by Croisis** [comment]
Yup, you are "burning" the plank itself as you use it, one hole at a time, which is why as I've said although not 100% accurate wording, the "burnout" flag and its description seems like a natural fit to me.
You can roughly tell how much space you have left to use but not with any kind of precision, hence the 1/4 steps fit IMHO.

**Comment by OnTheDouble** [comment]
Speaking of which [r]eloading with a stick would be a witty solution.

**Comment by Croisis** [comment]
Of that I'm not too sure TBH, the string will take a beating as well, we are talking about some random rough stick and you currently have 50 charges available to use. Effective reload is something I would not consider feasible IRL.
Anyway, I'll make a pull request with the flag and we'll see if there are any complaints about that solution or not.

**Comment by OnTheDouble** [comment]
UPDATE: As of today bow fire drill began taking damage with use. This is a satisfactory outcome I think although not the one I expected. =)
![image](https://user-images.githubusercontent.com/82908191/159362031-e961aff1-1865-4580-9225-11520f3b8822.png)
